packages/server/src/modules/google/google.auth.test.ts

```diff
@@ -0,0 +1,195 @@
+import request from 'supertest'
+import uuid from 'uuid/v4'
+import { Connection as DBConnection } from 'typeorm'
+import { mocked } from 'ts-jest/utils'
+
+import { createApp } from '@/apollo'
+import { connectToDatabase } from '@/db'
+import { AuthErrorCode } from '@/constants/auth.constants'
+import { Google, GoogleUser } from '@/modules/google/google.lib'
+import { User } from '@/modules/user/user.model'
+import {
+  Connection,
+  ConnectionService,
+} from '@/modules/connection/connection.model'
+import { Session } from '@/modules/session/session.model'
+
+jest.mock('@/modules/google/google.lib')
+const mockedGoogle = mocked(Google)
+
+let dbConnection: DBConnection
+const app = createApp()
+
+beforeAll(async () => {
+  dbConnection = await connectToDatabase()
+})
+
+beforeEach(async () => {
+  await dbConnection.synchronize(true)
+  jest.resetAllMocks()
+
+  mockedGoogle.getTokens.mockResolvedValue({
+    idToken: 'id_token',
+    token: 'the_token',
+    refreshToken: 'refresh_token',
+  })
+})
+
+afterAll(() => dbConnection.close())
+
+const assertLoggedIn = async (response: request.Response) => {
+  expect(response.header).toMatchObject({
+    'set-cookie': expect.arrayContaining([expect.stringContaining('token=')]),
+  })
+
+  const setCookies = response.header['set-cookie'] as string[]
+  const lastCookie = setCookies[setCookies.length - 1]
+
+  const token = /token=([\w\d-])+;/
+    .exec(lastCookie)![0]
+    .slice(6, -1)
+
+  expect(token).not.toBeNull()
+  expect(token).toMatch(
+    /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i,
```

Comment:
do we really need to test the format of the token?

Reply:
Not really

Reply:
It also does not hurt